sys-libs/libnbd/libnbd-1.23.3.ebuild

```diff
@@ -0,0 +1,119 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+RUST_OPTIONAL=1
+RUST_REQ_USE="rustfmt"
+PYTHON_COMPAT=( python3_{10..13} )
+
+inherit autotools bash-completion-r1 python-single-r1 rust
+
+MY_PV_1="$(ver_cut 1-2)"
+MY_PV_2="$(ver_cut 2)"
+[[ $(( ${MY_PV_2} % 2 )) -eq 0 ]] && SD="stable" || SD="development"
+
+DESCRIPTION="NBD client library in userspace"
+HOMEPAGE="https://gitlab.com/nbdkit/libnbd"
+SRC_URI="https://download.libguestfs.org/libnbd/${MY_PV_1}-${SD}/${P}.tar.gz"
+
+LICENSE="LGPL-2.1+"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~sparc ~x86"
+IUSE="examples fuse gnutls go ocaml python rust test"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+RESTRICT="!test? ( test )"
+
+# libxml2 - URI support
+RDEPEND="
+	dev-libs/libxml2
+	examples? (	dev-libs/glib
+			dev-libs/libev )
+	fuse? ( sys-fs/fuse:3 )
+	gnutls? ( net-libs/gnutls:= )
+	go? ( dev-lang/go )
+	ocaml? ( >=dev-lang/ocaml-4.03:=[ocamlopt] )
+	python? ( ${PYTHON_DEPS} )
+"
+DEPEND="
+	${RDEPEND}
+	test? (	sys-block/nbdkit[gnutls?]
+		net-libs/gnutls:=[tools]
+		ocaml? ( dev-ml/findlib[ocamlopt] )
+)
+"
+BDEPEND="dev-lang/perl
+	rust? ( ${RUST_DEPEND} )"
+
+pkg_setup() {
+	if use rust; then
+		rust_pkg_setup
+	fi
+	if use python; then
+		python_setup
+	fi
+}
+
+src_prepare() {
+	default
+
+	if ! use examples; then
+		sed -i Makefile.am -e '/^[[:blank:]]*examples[[:blank:]]*\\$/d' || die
+		sed -i Makefile.am -e '/^[[:blank:]]*ocaml\/examples[[:blank:]]*\\$/d' || die
+		sed -i Makefile.am -e '/^[[:blank:]]*rust\/examples[[:blank:]]*\\$/d' || die
+	fi
+	if ! use fuse; then
+		sed -i Makefile.am -e '/^[[:blank:]]*fuse[[:blank:]]*\\$/d' || die
+	fi
+	if ! use go; then
+		sed -i Makefile.am -e '/^[[:blank:]]*golang[[:blank:]]*\\$/d' || die
+	fi
+	if ! use ocaml; then
+		sed -i Makefile.am -e '/^[[:blank:]]*ocaml[[:blank:]]*\\$/d' || die
+		sed -i Makefile.am -e '/^[[:blank:]]*ocaml\/tests[[:blank:]]*\\$/d' || die
+	fi
+	if ! use python; then
+		sed -i Makefile.am -e '/^[[:blank:]]*python[[:blank:]]*\\$/d' || die
+# sh tests not support if python binding not built
+		sed -i Makefile.am -e '/^[[:blank:]]*sh[[:blank:]]*\\$/d' || die
+	fi
+	if ! use rust; then
+		sed -i Makefile.am -e '/^[[:blank:]]*rust[[:blank:]]*\\$/d' ||die
+	fi
+
+	# Not in portage
+	sed -i Makefile.am -e '/[[:blank:]]*ublk[[:blank:]]*\\$/d' || die
+
+	# Broken under sandbox.
+	cat <<-EOF > lib/test-fork-safe-execvpe.sh || die
+	#!/bin/sh
+	:
+	EOF
+
+	eautoreconf
```

Comment:
can we avoid autoreconf by editing the `Makefile.in`?  if not, that's OK too, but I suspect it might be possible.

Reply:
It's generally preferable to do `eautoreconf`, at least for patches to always contain the configure.ac et. al hunks so that if in future we _do_ need to `eautoreconf`, we don't have to reconstruct the changes.